### PR TITLE
Fwmt 1044 pass cancel case to comet

### PR DESCRIPTION
**Pull request by SteveScorfield**

### CODE REVIEW

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java [hunk 1]
```diff
@@ -73,14 +73,14 @@ private void auth() throws GatewayException {
     }
   }
 
-  public void sendCreateJobRequest(CaseRequest caseRequest, String caseId) throws GatewayException {
+  public <A> void sendCreateJobRequest (A caseRequest, String caseId) throws GatewayException {
```
Comment (JacobHarrisonQA) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java]: consider renaming this method name. It doesn't only send create jobs now.

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/service/JobService.java [hunk 2]
```diff
@@ -10,4 +10,6 @@
   void cancelJob(CancelFieldWorkerJobRequest cancelRequest) throws GatewayException;
 
   void convertAndSendCreate(CreateFieldWorkerJobRequest jobRequest) throws GatewayException;
+
+  void convertAndSendCreateCancel(CancelFieldWorkerJobRequest cancelJobRequest) throws GatewayException;
```
Comment (JacobHarrisonQA) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/service/JobService.java]: rename this also

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/service/impl/JobServiceImpl.java [hunk 3]
```diff
@@ -46,4 +47,13 @@ public void convertAndSendCreate(CreateFieldWorkerJobRequest jobRequest) throws
     cometRestClient.sendCreateJobRequest(caseRequest, String.valueOf(jobRequest.getCaseId()));
     gatewayEventManager.triggerEvent(String.valueOf(jobRequest.getCaseId()), COMET_CREATE_ACK, LocalTime.now());
   }
+
+  public void convertAndSendCreateCancel(CancelFieldWorkerJobRequest cancelJobRequest) throws GatewayException {
+    final CometConverter cometConverter = cometConverters.get(cancelJobRequest.getCaseId());
```
Comment (JacobHarrisonQA) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/service/impl/JobServiceImpl.java]: Do not think we need this. It's not using the named component feature of the comet converter where it grabs the survey type and will use the converter of the named survey type. take a look at the create job for more clarification.

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java (hunk 1 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java]: Name changed to be more generic

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/service/JobService.java (hunk 2 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/service/JobService.java]: Updated name

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/service/impl/JobServiceImpl.java (hunk 3 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/service/impl/JobServiceImpl.java]: Updated as requested

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/config/GatewayEventsConfig.java [hunk 4]
```diff
@@ -11,13 +11,14 @@
   public static final String CANONICAL_CANCEL_RECEIVED = "Canonical - Cancel Job Received";
   public static final String COMET_CREATE_SENT = "Comet - Create Job Request";
   public static final String COMET_CREATE_ACK = "Comet - Create Job Acknowledged";
-
```
Comment (JacobHarrisonQA) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/config/GatewayEventsConfig.java]: delete space

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java [hunk 5]
```diff
@@ -73,14 +73,14 @@ private void auth() throws GatewayException {
     }
   }
 
-  public void sendCreateJobRequest(CaseRequest caseRequest, String caseId) throws GatewayException {
+  public <A> void sendAllRequestTypes(A caseRequest, String caseId) throws GatewayException {
```
Comment (JacobHarrisonQA) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java]: name sendRequest

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/service/JobService.java [hunk 6]
```diff
@@ -10,4 +10,6 @@
   void cancelJob(CancelFieldWorkerJobRequest cancelRequest) throws GatewayException;
 
   void convertAndSendCreate(CreateFieldWorkerJobRequest jobRequest) throws GatewayException;
+
+  void convertAndSendCancelledJob(CancelFieldWorkerJobRequest cancelJobRequest) throws GatewayException;
```
Comment (JacobHarrisonQA) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/service/JobService.java]: name convertAndSendCancel

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/service/impl/JobServiceImpl.java [hunk 7]
```diff
@@ -14,8 +15,7 @@
 import java.time.LocalTime;
 import java.util.Map;
 
-import static uk.gov.ons.census.fwmt.jobservice.config.GatewayEventsConfig.COMET_CREATE_ACK;
-import static uk.gov.ons.census.fwmt.jobservice.config.GatewayEventsConfig.COMET_CREATE_SENT;
+import static uk.gov.ons.census.fwmt.jobservice.config.GatewayEventsConfig.*;
```
Comment (JacobHarrisonQA) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/service/impl/JobServiceImpl.java]: no wildcards

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/config/GatewayEventsConfig.java (hunk 4 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/config/GatewayEventsConfig.java]: Space deleted

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/service/JobService.java (hunk 6 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/service/JobService.java]: Name changed

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/service/impl/JobServiceImpl.java (hunk 7 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/service/impl/JobServiceImpl.java]: Wildcards removed

**Pull request merged**